apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/tasks/OperationTasksBuilder.java

```diff
@@ -19,46 +20,47 @@
 @AllArgsConstructor
 public class OperationTasksBuilder {
   @Getter(AccessLevel.NONE)
-  private final OperationTaskFactory<? extends OperationTask> taskFactory;
+  private final OperationTaskFactory<? extends OperationTask<?>> taskFactory;
 
   protected final TablesClient tablesClient;
 
-  private List<OperationTask> prepareTableOperationTaskList(JobConf.JobTypeEnum jobType) {
-    List<OperationTask> taskList = new ArrayList<>();
-    for (TableMetadata metadata : tablesClient.getTables()) {
-      log.info("metadata: {}", metadata);
-      try {
-        taskList.add(taskFactory.create(metadata));
-      } catch (Exception e) {
-        throw new RuntimeException(
-            String.format(
-                "Cannot create operation task from TableMetadata: %s given job type: %s",
-                metadata, jobType),
-            e);
-      }
-    }
-    return taskList;
+  private List<OperationTask<?>> prepareTableOperationTaskList(JobConf.JobTypeEnum jobType) {
+    List<TableMetadata> tableMetadataList = tablesClient.getTableMetadataList();
+    log.info("Fetched metadata for {} tables", tableMetadataList.size());
```

Comment:
Does this need to be logged in info? Or is this more for debugging?

Reply:
We use this information internally quite often, it's cheap to log it